edxval/tests/test_views.py

```diff
@@ -1101,3 +1102,33 @@ def test_update_hls_encodes_for_video(self):
         self.assertEqual(actual_encoded_video.url, expected_data['encode_data']['url'])
         self.assertEqual(actual_encoded_video.file_size, expected_data['encode_data']['file_size'])
         self.assertEqual(actual_encoded_video.bitrate, expected_data['encode_data']['bitrate'])
+
+
+class CourseTranscriptsDetailViewTest(APIAuthTestCase):
+    """
+    CourseTranscriptsDetailView Tests.
+    """
+    def setUp(self):
+        """
+        Tests setup.
+        """
+        self.base_url = 'course-transcripts'
+        super().setUp()
+
+    def test_successful_response(self):
+        """
+        Test succesful response from view
+        """
+        with patch(
+            'edxval.views.get_transcript_details_for_course'
+        ) as mock_transcript_details:
+            # Simulate a return value when the function is called.
+            mock_transcript_details.return_value = {}
+            course_id = 'course-v1:edx+1+2023_05'
+            url = reverse(self.base_url, args=[course_id])
+            response = self.client.get(url)
+
+            # Verify the function was called once with course_id
+            mock_transcript_details.assert_called_once_with(course_id)
+
+            self.assertEqual(response.status_code, status.HTTP_200_OK)
```

Comment:
nit: You should not mock the internal util here. Instead, setup the required data using factories and verify that view is getting the correct data as expected.

Reply:
@BrandonHBodine Hi, this was not addressed. I would suggest creating a followup PR to address this.

Reply:
Well, not a requirement for release but important from unit testing point of view. 
I agree that get_transcript_details_for_course has been tested in edxval/tests/test_api.py but the view is utilizing that util, so we have to ensure that view is working as expected by adding some test data. It is not redundant. Both are testing different parts of the code.